### PR TITLE
Add dummy app

### PR DESCRIPTION
**Pull request by ahangarha**
### Summary

Added a dummy app based on https://github.com/shakacode/react_on_rails_demo_ssr_hmr/pull/55.

### Pull Request checklist
- [x] Add/update test to cover these changes
- [x] Update documentation (will be added after initial approval)
- [x] Update CHANGELOG file (I don't think this should be included in Changelog)

### Other Information

Besides this, I have reorganized our specs so we can easily control running each set of specs. After finalization of this PR, some extra changes might be needed to be added to the old specs (ex., renaming `spec_bc`sufix for backward compatibility specs)

Todo:
- [ ] The CI for the dummy app test runs only with Ruby 3.1.2. Should we have a matrix of ruby versions for it? What about the Node and Rails versions?
- [x] Check whether the rake task related to the dummy app can be improved. If not, I should remove it. Update: Fixed

### CODE REVIEW

File: spec/dummy/README.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
+# React on Rails Demo With SSR, HMR fast refresh, and TypeScript
+
```
Comment (justin808) [inline spec/dummy/README.md:2]: does this file make sense in the spec dummy app?

File: spec/dummy/.yarnrc.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+nodeLinker: node-modules
```
Comment (justin808) [inline spec/dummy/.yarnrc.yml:1]: Is a .yarn directory appropriate? or should this be git ignored?

File: spec/dummy/app/javascript/bundles/HelloWorld/components/HelloWorld.mdx [hunk 3]
```diff
@@ -0,0 +1,4 @@
+### Hello, {props.name === '' ? 'World' : props.name} from MDX!
```
Comment (justin808) [inline spec/dummy/app/javascript/bundles/HelloWorld/components/HelloWorld.mdx:1]: Adding mdx here makes no sense. It's one more thing that will break in the future.

**Review by justin808 — CHANGES_REQUESTED** [review]
Be sure to do a self-review.

I want to ship v7 and I don't want this blocking.

File: spec/dummy/README.md (hunk 1 above)
Comment (ahangarha) [reply spec/dummy/README.md]: Updated so that it just says this dummy app is based on RORDSH app.

File: CHANGELOG.md [hunk 4]
```diff
@@ -51,8 +51,8 @@ _Please add entries here for your pull requests that are not yet released._
 - Remove deprecated `check_yarn_integrity` from `Shakapacker::Configuration` [PR SP288](https://github.com/shakacode/shakapacker/pull/288) by [ahangarha](https://github.com/ahangarha).
 
 ### Added
-- All standard Webpack entries with the camelCase format are now supported in `shakapacker.yml` in snake_case format. [PR276](https://github.com/shakacode/shakapacker/pull/276) by [ahangarha](https://github.com/ahangarha).    
-
+- All standard Webpack entries with the camelCase format are now supported in `shakapacker.yml` in snake_case format. [PR276](https://github.com/shakacode/shakapacker/pull/276) by [ahangarha](https://github.com/ahangarha).
+- Add dummy app [PR299](https://github.com/shakacode/shakapacker/pull/299) by [ahangarha](https://github.com/ahangarha).
```
Comment (justin808) [inline CHANGELOG.md:55]: remove

File: spec/dummy/.nvmrc [hunk 5]
```diff
@@ -0,0 +1 @@
+14
```
Comment (justin808) [inline spec/dummy/.nvmrc:1]: not sure if this is needed.

File: spec/dummy/LICENSE [hunk 6]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (justin808) [inline spec/dummy/LICENSE:1]: not needed

**Review by justin808 — APPROVED** [review]
couple small changes, then merge.

**Review by justin808 — APPROVED** [review]
couple small changes, then merge.

**Pull request merged**